### ISSUE TITLE
TFT 70 Trouble Installing

### ISSUE THREAD

**Issue by RavaszEmber** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1280]
I'm having trouble installing the latest firmware on my TFT 70.

I downloaded "BIGTREETECH-TouchScreenFirmware-master" and navigated to Copy to "SD Card root directory to update - Unified Menu Material theme".
I then moved "config.ini" "BIGTREE_TFT70_V3.0.26.x" and the "TFT70" folder to my SD card. These files are in the attached zip file.
[Files.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/5579110/Files.zip)

I then loaded this to my TFT70 and powered on my printer. When updating I had errors that icons were missing, namely "S_SAVE.bmp" and "S_LOAD.bmp".
When updating configuration, I had errors that keywords were invalid, namely "mesh_min_color:7" and "mesh_max_color:2"

After the display finished updating I was not prompted with a calibration screen and I was unable to use the touchscreen.

**Comment by oldman4U** [comment]
@bigtreetech 

Next user struggling with outdated binaries.

When will you release the PRs!!??

**Comment by oldman4U** [comment]
Hi.

You did everything right, but unfortunately the owner of this repository is - for an unknown reason - not able to merge even the most important PR's in an acceptable timeframe. Sorry

Please use the repository below, instead of the official one.

Thank you

https://github.com/radek8/BIGTREETECH-TouchScreenFirmware/tree/updateBIN

**Comment by RavaszEmber** [comment]
Thanks old man, that update worked great. However, I am still unable to use the touchscreen. I assume I have to calibrate it but I can't get to the calibration menu without the touchscreen in the first place. Is there a way to run the calibration command from the SD card after updating?

Thanks!

**Comment by kisslorand** [comment]
Make an empty "reset.txt" file on your card and reboot TFT with the card inserted.

**Comment by oldman4U** [comment]
https://github.com/oldman4U/BIGTREETECH-TouchScreenFirmware#troubleshooting

Please help the community and close the ticket once you do not need it anymore.

Thank you

**Comment by RavaszEmber** [comment]
Thanks for the help, I just have one last issue. My screen is stuck reporting that no printer is connected. I checked that the 5 pin TFT connector is connected to my SKR 1.3 and checked the that the baudrate for the display is the same as the baudrate I have in Marlin (115200).
I'm guessing the problem is my serial pin isn't defined for the TFT connector, but I'm not sure how to check this. Here's my relevant Marlin code:
![image](https://user-images.githubusercontent.com/28627787/99916911-03508280-2cdb-11eb-9d56-5cf52568a88b.png)

**Comment by RavaszEmber** [comment]
Should SERIAL_PORT be set to 0 and SERIAL_PORT_2 be set to -1?

**Comment by oldman4U** [comment]
Yes, I would change them, so the connection to the TFT is the one with more priority, even the two SHOULD be the same......

Send a picture which shows the connection from the TFT to the mainboard.

**Comment by RavaszEmber** [comment]
Bit of a mess but hope this helps.
![image](https://user-images.githubusercontent.com/28627787/99917218-dc934b80-2cdc-11eb-8999-ccfc01e1c240.png)
![image](https://user-images.githubusercontent.com/28627787/99917237-f2087580-2cdc-11eb-8314-5c427383d655.png)
The pin on the far left of the tft header on the mainboard is the single reset pin. I tried this the other way around just to make sure it wasn't a connection issue.

**Comment by oldman4U** [comment]
Thank you for the picture. And what you wrote is exactly what I wanted to find out;-)

For the Touchscreen connection 3 things are important.

1. cable working and connected right
2. serial port defined for TFT connection
3. baud rate the same on TFT and mainboard

The No Printer Connected simply means that the serial port of the TFT can not "find" the mainboard. Check the cable with a multimeter.... last hope I must say.;-(

**Comment by RavaszEmber** [comment]
I checked the cable for continuity and it all checks out (I'd be baffled if they were somehow defective).
I've been putting off updating my mainboard because of the pain that is to do. It sounds like I should change SERIAL_PORT to 0 and SERIAL_PORT_2 to -1 and hope that works. I always doubt that the hardware would be an issue. Maybe I have the wrong display type set in marlin or something? I've attached my Marlin configuration.h in case you're able to spot an issue there. If not I'll update the mainboard only changing SERIAL_PORT and SERIAL_PORT_2.

Thanks for the help!


[Marlin_Configuration.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/5580676/Marlin_Configuration.zip)

**Comment by oldman4U** [comment]
No Ptinter Attached error only the previous mentioned parameters are
relevant. All other parameters are Marlin mode related.

Benjamin Molnar <notifications@github.com> schrieb am Mo. 23. Nov. 2020 um
01:18:

**Comment by Doyle4** [comment]
Try changing the Baudrate in the settings, even if they match in the firmware, its been known to still not work and need changing, it maybe set to 115000 at the moment, try 25000 or vice versa.

**Comment by oldman4U** [comment]
@Doyle4 

You mean to change the baud rate back and forth between two values, where the one which finally works has been selected the time the TFT has been started, but it even did not work having the correct setting enabled???

Do you experience this?

**Comment by Doyle4** [comment]
I have had a screen that would only work with 115000 when the printer was set to 250000, how it worked I have no idea.
Wrong naming for types maybe?

**Comment by oldman4U** [comment]
And you know exactly which screen and which firmware version in combination with which mainboard and firmware? Was it BTT stuff?

**Comment by radek8** [comment]
Note: Ensure that following options are enabled in Marlin firmware

  M115_GEOMETRY_REPORT (in Configuration_adv.h)
  M114_DETAIL  (in Configuration_adv.h)
  REPORT_FAN_CHANGE (in Configuration_adv.h)
  EMERGENCY_PARSER (in Configuration_adv.h)
  SERIAL_FLOAT_PRECISION 4 (in Configuration_adv.h)
  HOST_ACTION_COMMANDS  (in Configuration_adv.h)

 For printing from onboard SD

  SDSUPPORT (in Configuration.h)
  AUTO_REPORT_TEMPERATURESS (in Configuration_adv.h)
  AUTO_REPORT_SD_STATUS (in Configuration_adv.h)
  LONG_FILENAME_HOST_SUPPORT (in Configuration_adv.h)
  SDCARD_CONNECTION ONBOARD (in Configuration_adv.h)

**Comment by oldman4U** [comment]
> Note: Ensure that following options are enabled in Marlin firmware
> 
> M115_GEOMETRY_REPORT (in Configuration_adv.h)
> M114_DETAIL (in Configuration_adv.h)
> REPORT_FAN_CHANGE (in Configuration_adv.h)
> EMERGENCY_PARSER (in Configuration_adv.h)
> SERIAL_FLOAT_PRECISION 4 (in Configuration_adv.h)
> HOST_ACTION_COMMANDS (in Configuration_adv.h)
> 
> For printing from onboard SD
> 
> SDSUPPORT (in Configuration.h)
> AUTO_REPORT_TEMPERATURESS (in Configuration_adv.h)
> AUTO_REPORT_SD_STATUS (in Configuration_adv.h)
> LONG_FILENAME_HOST_SUPPORT (in Configuration_adv.h)
> SDCARD_CONNECTION ONBOARD (in Configuration_adv.h)

@traffic-light Could you please add this information to your ReadMe first document. Thank you

**Comment by radek8** [comment]
According to the attached photo, the connector is connected upside down.

**Comment by oldman4U** [comment]
Any news?

**Comment by RavaszEmber** [comment]
I haven't gotten a chance to update the Marlin code on my mainboard yet. I'll try changing the mainboard's baudrate to 250000, SERIAL_PORT to 0, SERIAL_PORT_2 to -1 and I'll check that the settings mentioned by radek8 are enabled.
I double checked my wiring based on radek8's comment and the way I have it wired is correct. This can be seen with the attached picture that shows the pin labels on the bottom of my physical board. I'm not sure why my pinout has reset in a different spot than in the skr 1.3 pinout. When the connection is flipped the screen does not turn on. I can conclude the physical connection is correct.
![image](https://user-images.githubusercontent.com/28627787/100118851-37938280-2e44-11eb-90ff-f36fe4140919.png)


I have tried changing the buadrate to every preset available and the "printer not attached" issue was never solved with any baudrate. 
If my understanding is correct, the only things needed for the display to attach to the board is wired TFT connection and the same baudrate. I'm not sure why this issue persists, perhaps I'm looking over something simple? It seems to me that I set everything required to connect the screen to the printer. I'm hoping a Marlin update will fix this.

**Comment by RavaszEmber** [comment]
I just looked at it again and my connector is in backwards, but the right way leaves me with a blank screen (but a light on in the back) and the wrong way leaves me with an operating screen. I'm very confused. Pulling out jumpers and the multimeter.

**Comment by RavaszEmber** [comment]
Alright at this point I'm just baffled. Why does my screen light up when I connect RS232 + and - to RST and R0 (respectively) but the screen does not light up when I connect RS232 + and - to 5V and G respectively?

**Comment by radek8** [comment]
You got everything wrong

**Comment by RavaszEmber** [comment]
@radek8 care to elaborate? When the connector is placed backwards my screen turns on, when it's in the right way the screen remains off.

**Comment by radek8** [comment]
Connect the serial cable via RS232 correctly rotated.
Connect cables EXP1 and EXP2 as shown.
You used EXP3, so the print gets into the reset loop
You must not use the EXP3 connector

![image](https://user-images.githubusercontent.com/46979052/100130220-c0c9aa00-2e82-11eb-81bc-cdd228efbca9.png)

**Comment by radek8** [comment]
![image](https://user-images.githubusercontent.com/46979052/100132093-36cf1080-2e85-11eb-8810-f5b8c769d915.png)

**Comment by RavaszEmber** [comment]
![image](https://user-images.githubusercontent.com/28627787/100131853-965ff880-2e52-11eb-8416-33e88f11d864.png)
That worked, thank you!
One last thing, when I switch to Marlin Mode now the screen remains blank. Am I not able to switch between the two? (pic below)
![image](https://user-images.githubusercontent.com/28627787/100132225-1c7c3f00-2e53-11eb-96ab-57863b8608de.png)

**Comment by radek8** [comment]
Marlin mod communicates after EXP1 and EXP2
The cables must be connected correctly.
In Marlin you have to enable (delete //)
#define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER

all other display drivers including
// # define CR10_STOCKDISPLAY
you have to comment

**Comment by RavaszEmber** [comment]
Thanks a bunch!

TLDR: Wrong connections 😛

**Comment by oldman4U** [comment]
Benjamin, which instructions did you follow initially, to connect the TFT?
What made you believe the cables are connected right?

Benjamin Molnar <notifications@github.com> schrieb am Di. 24. Nov. 2020 um
19:03:

**Comment by radek8** [comment]
Unnecessary mistakes due to inattention.
You're lucky you didn't destroy anything

**Comment by RavaszEmber** [comment]
I had originally connected EXP 3 so I could use the original marlin display when I first got it. I guess I flipped the TFT connection to the reverse because the correct way led to a blank screen. I overlooked the EXP connections and thought it was an issue with the serial connection from our conversation.

**Comment by radek8** [comment]
Did you fix the Marlin configuration?
Is the Marlin simulation already working?

**Comment by RavaszEmber** [comment]
I did not @radek8. I think I'll be able to make the changes based off your comments. I'll do it when I have time and let you know if I run into any problems.

**Comment by radek8** [comment]
OK,
don't forget the settings I sent earlier.

**Comment by oldman4U** [comment]
Usually a manual or YouTube instructions are used to get the information
needed to install something like a display. Or did you just try it and once
it did not work the next was to open a ticket here? Have you seen and read
the ReadMe at the front page of this repository?

Benjamin Molnar <notifications@github.com> schrieb am Di. 24. Nov. 2020 um
19:22:

**Comment by RavaszEmber** [comment]
@oldman4U I did reference a couple YouTube videos before installing. The original motivation for this thread was due to the errors I encountered when updating the display's firmware from the BTT Github. Having errors with the official github code caused me to open this thread (I'm not sure how I would have resolved that if I didn't make a thread). Since I still had issues after that I continued the thread until it was pointed out that connections were overlooked.

**Comment by oldman4U** [comment]
Ok. Thank you

Happy printing

Benjamin Molnar <notifications@github.com> schrieb am Di. 24. Nov. 2020 um
19:41:

**Comment by billyd60** [comment]
Hi

I am trying to connect the tft 70 to an archim2.  My serial port pins (on the main board) are not in the same order as the connector, e.g. the rst pin is in the center and the ground and + are reversed. Anyway I re-ordered the pins in the connector (that will attach to the main board) to match but I have a question. Should tx from the display go to rx on the board and vice versa? That's normally what you do in serial connections but I wanted to make sure.

-Bill

**Comment by oldman4U** [comment]
Check the documentation of SKR board and your TFT where the connection works without twisting wires.

**Comment by billyd60** [comment]
> Check the documentation of SKR board and your TFT where the connection works without twisting wires.

I'm using the archim2. I've isolated the issue to the serial port. Can't seem to get the archim2 serial port functioning in marlin2

**Comment by radek8** [comment]
You must select the correct port number in Marin

**Comment by billyd60** [comment]
> You must select the correct port number in Marin

Thanks I heard back from ultimachine and I needed to define serial_port -1 and serial_port_2 0

Of course BAUD had to match on both as well I used 250000 and that resolved the serial connection on J28 (on the Archim2 board).

Now I have the BTT TFT70 v3 working via serial, but I can't get the 12864 marlin emulator working through the two ribbon cable connectors. When I plug those in to SC-1 and SC-2 on the Archim2 board the BTT TFT70 gets stuck at the choosing screen (Touch screen or Marlin emulator) and will not respond. I've verified the cables are all correct, pin1 to pin1 but nothing works until I disconnect those cables. I suppose I could just use the touchscreen and forget about Marlin emulation but it would be nice to have everything working as it should.

**Comment by radek8** [comment]
Depending on the connection, it seems that the Archim2 board has a different connection of the SC1 and SC2 connectors, opposite the EXP1 and EXP2 connectors on the TFT display. Your old display works because it has the same atypical connection as Archim2

![image](https://user-images.githubusercontent.com/46979052/114769138-176c4d80-9d6a-11eb-912a-5df59208f7ab.png)

![image](https://user-images.githubusercontent.com/46979052/114769480-8e094b00-9d6a-11eb-92bd-7ba5a39da8f2.png)

**Comment by radek8** [comment]
The pins defined are, if you select the right display driver, it could work ...

#define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER

![image](https://user-images.githubusercontent.com/46979052/114771903-6d8ec000-9d6d-11eb-8f20-97c8b7ecce99.png)

**Comment by billyd60** [comment]
So are you saying the pins are wrong or right? How do I fix it?

**Comment by radek8** [comment]
And the Marlin mod works with
#define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER ?

**Comment by radek8** [comment]
I don't have your board and I can't try it.
But when I look at the pin definition and wiring diagram, it looks like 2 pins have been swapped in the pin definition file
#define BEEPER_PIN 23 // D24 PA15_CTS1
#define LCD_PINS_ENABLE 24 // D23 PA14_RTS1

![image](https://user-images.githubusercontent.com/46979052/114780281-771d2580-9d77-11eb-8837-2050de711b68.png)

**Comment by billyd60** [comment]
I just discovered something strange on the archim. My sc1 connector has 3.3v across 9 and 10 which is supposed to be on sc2. And my sc2 has 5v across pins 1 and 2 which should be 9 and 10 on sc1!

So it appears they are not only marked wrong but one is upside down.

**Comment by radek8** [comment]
Yes SC1 has a 5V supply
and comes to connect with EXP1 on the TFT
CS2 has a 3.3V power supply but the pin is not connected to EXP1.
Check that the connection is correct

**Comment by radek8** [comment]
![image](https://user-images.githubusercontent.com/46979052/114783637-4c809c00-9d7a-11eb-8943-ff2b4b173913.png)

**Comment by billyd60** [comment]
Yes and in that picture my sc2 is apparently where sc1 is supposed to be and pin 1 is the upper left. And sc2 is where sc1 is supposed to be and pin 1 is the upper right.

**Comment by billyd60** [comment]
How in the world did that happen? Is it something I. The pins file? Seems no because you can't change ground with firmware can you?

**Comment by radek8** [comment]
No, the power pins are set and cannot be rotated. But in the pin file you can give individual pins of the processor different functions ...

**Comment by radek8** [comment]
![image](https://user-images.githubusercontent.com/46979052/114785612-3c1df080-9d7d-11eb-8592-7108ec9129c3.png)

**Comment by radek8** [comment]
Disconnect the serial cable. Only connect EXP1 - SC1.
The TFT display should boot if the cable is oriented correctly, the touch screen will report the printer is not connected, Marlin mode should display the initial screen but will not respond to knob rotation.
Can you try it?

**Comment by billyd60** [comment]
In your picture I get 3.3 volts on the right hand most pins of the top connector labelled on the board as sc1 with the ground on the top pin.

I get 5v on the left hand set of pins of the bottom connector in your photo labelled sc2 with ground on the bottom pin.

Crazy

**Comment by radek8** [comment]
You measure 3.3V if the processor pin is at a high level, that's fine.
The important thing is where you measure the 5V power supply

**Comment by radek8** [comment]
Either the manufacturer has poor documentation, or there are several HW variants of this board.

**Comment by radek8** [comment]
Try the connection I suggested above

**Comment by radek8** [comment]
It is late I am going to sleep

**Comment by billyd60** [comment]
Ok I I tried the one cable. Screen does not power up

**Comment by billyd60** [comment]
I got it working! I pulled the plastic cases off the archim board around sc1 and sc2. Plugged them both in but upside down at the archim2 board. I did connect sc1 to exp1 and sc2 to exp2. I also plugged in the serial connection and now it works!
